src/controllers/wishlistController.ts

```diff
@@ -0,0 +1,104 @@
+import { Request, Response } from 'express';
+import { sendErrorResponse } from '../helpers/helper';
+import { Wishlist } from '../database/models/wishlist';
+import { Product } from '../database/models/Product';
+import { Size } from '../database/models/Size';
+import { sendInternalErrorResponse } from '../validations';
+
+interface UserInterface {
+  id: string;
+  name: string;
+  email: string;
+}
+
+export const addToWishlist = async (req: Request, res: Response) => {
+  const { productId } = req.body;
+  const user = req.user as UserInterface;
+
+  try {
+    const product = await Product.findOne({
+      where: { id: productId },
+    });
+    if (!product) {
+      return sendErrorResponse(res, 'Product not found');
+    }
+    const size = await Size.findOne({
+      where: { productId: productId },
+    });
+    if (!size) {
+      return sendErrorResponse(res, 'No sizes found !');
+    }
+    //check if the product is already added in the wishlist
+    const itemExist = await Wishlist.findOne({
+      where: {
+        userId: user.id,
+        productId: productId,
+      },
+    });
+    if (itemExist) {
+      return res.status(400).json({
+        ok: false,
+        message: 'Product already added to wishlist',
+      });
+    }
+    const wishlistItem = await Wishlist.create({
+      userId: user.id,
+      productId: productId,
+      name: product.name,
+      price: size.price,
+      images: product.images,
+    });
+
+    return res.status(201).json({
+      ok: true,
+      message: 'Product added to wishlist successfully',
+      data: wishlistItem,
+    });
+  } catch (err) {
+    console.log(err);
```

Comment:
Remove console.log, use logger.error instead

Reply:
I was using that when i had a bug ,but it's no longer there